### PR TITLE
Add template specialization for Convert<boost::array> utility

### PR DESCRIPTION
**Pull request by jamoralp**

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/soss/37/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/soss/38/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/soss/41/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/soss/44/)

File: packages/core/include/soss/utilities.hpp [hunk 1]
```diff
@@ -190,8 +190,8 @@ struct ContainerConvert
     vector.resize(size);
   }
 
-  template<typename T, std::size_t N>
-  static void container_resize(std::array<T, N>& /*array*/, std::size_t /*size*/)
+  template<template <typename Kind, std::size_t S> class Array, typename T, std::size_t N>
```
Comment (richiware) [inline packages/core/include/soss/utilities.hpp]: I think there are duplicated typenames. Kind = T and S = N. Can we join them or maybe use a `static_assert` to detect they are different?

File: packages/core/include/soss/utilities.hpp [hunk 2]
```diff
@@ -228,11 +228,11 @@ struct Convert<std::vector<ElementType, Allocator>>
     std::numeric_limits<typename std::vector<ElementType, Allocator>::size_type>::max()> { };
 
 //==============================================================================
-template<typename ElementType, std::size_t N>
-struct Convert<std::array<ElementType, N>>
+template<template <typename Kind, std::size_t S> class Array, typename ElementType, std::size_t N>
```
Comment (richiware) [inline packages/core/include/soss/utilities.hpp]: I think there are duplicated typenames. Kind = ElementType and S = N. Can we join them or maybe use a `static_assert` to detect they are different?

Comment (jamoralp) [reply packages/core/include/soss/utilities.hpp]: Now typenames are unique for the template specialization.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/soss/46/)

**Pull request merged**